public_html/script.js

```diff
@@ -1106,55 +1106,29 @@ function refreshHighlighted() {
 		highlighted = Planes[HighlightedPlane];
 	}
 
-	// no highlighted plane
-	if (!highlighted) {
-		$('#highlighted_infoblock').hide();
-		return;
-	}
-
-	$('#highlighted_infoblock').show();
-
-	// Get info box position and size
 	var infoBox = $('#highlighted_infoblock');
-	var infoBoxPosition = infoBox.position();
-	if (typeof infoBoxOriginalPosition.top === 'undefined') {
-		infoBoxOriginalPosition.top = infoBoxPosition.top;
-		infoBoxOriginalPosition.left = infoBoxPosition.left;
-	} else {
-		infoBox.css("left", infoBoxOriginalPosition.left);
-		infoBox.css("top", infoBoxOriginalPosition.top);
-		infoBoxPosition = infoBox.position();
+
+	// no highlighted plane or in process of removing plane
+	if (!highlighted || !highlighted.marker) {
+		infoBox.fadeOut();
+		return;
 	}
-	var infoBoxExtent = getExtent(infoBoxPosition.left, infoBoxPosition.top, infoBox.outerWidth(), infoBox.outerHeight());
+	infoBox.fadeIn(100);
 
-	// Get map size
 	var mapCanvas = $('#map_canvas');
-	var mapExtent = getExtent(0, 0, mapCanvas.width(), mapCanvas.height());
-
-	var marker = highlighted.marker;
 	var markerCoordinates = highlighted.marker.getGeometry().getCoordinates();
     var markerPosition = OLMap.getPixelFromCoordinate(markerCoordinates);
-
-	// Check for overlap
-	//FIXME TODO: figure out this/remove this check
-	if (isPointInsideExtent(markerPosition[0], markerPosition[1], infoBoxExtent) || true) {
-		// Array of possible new positions for info box
-		var candidatePositions = [];
-		candidatePositions.push( { x: 40, y: 80 } );
-		candidatePositions.push( { x: markerPosition[0] + 20, y: markerPosition[1] + 60 } );
-
-		// Find new position
-		for (var i = 0; i < candidatePositions.length; i++) {
-			var candidatePosition = candidatePositions[i];
-			var candidateExtent = getExtent(candidatePosition.x, candidatePosition.y, infoBox.outerWidth(), infoBox.outerHeight());
-
-			if (!isPointInsideExtent(markerPosition[0],  markerPosition[1], candidateExtent) && isPointInsideExtent(candidatePosition.x, candidatePosition.y, mapExtent)) {
-				// Found a new position that doesn't overlap marker - move box to that position
-				infoBox.css("left", candidatePosition.x);
-				infoBox.css("top", candidatePosition.y);
-			}
-		}
-	}
+        var x = markerPosition[0] + 20;
+        var y = markerPosition[1] + 60;
+        var w = infoBox.outerWidth() + 20;
+        var h = infoBox.outerHeight();
+        if (x > mapCanvas.width() - w) {
+                x -= w + 20;
+        }
+        if (y > mapCanvas.height() - h) {
+                y -= h;
+        }
+        infoBox.animate({ left: x, top: y }, 500);
```

Comment:
I like animate because the infoBox appears where you last saw it-- there can be only one!

Reply:
I think it makes more sense to think of each marker having its own infoBox with its own flight info. As a user, the animation makes it confusing to hover over a plane, expecting to see its infoBox, and instead the infoBox appears somewhere else on the page and then scoots over to the cursor. Without the animation, the info simply appears where expected.

Reply:
I'm going to agree with @thornjad on this one. The fade in is slick, but the animate/moving is pretty jarring, especially on the initial page load where it just comes zooming in from the top left.

Reply:
So the page has to have state, to remember if the infoBox was already on the page?
Or does it always jump, even at the edge when it has to slide over a bit to stay on the page?
You were fine with jumping around, but not when it first appears?
Where else would the infoBox be, if never before shown, except for the top left?
I don't see animate as jarring, but jumping aound when it reaches the edge is.
Recall that there is only one infoBox, even in the user's perspective. Should it not appear where you last saw it? The user is looking at the marker under the mouse.

Reply:
Well I'm dumb and accidentally deleted my comment, but I agree with @thornjad. The fade in/out is slick, but the animation is jarring especially on initial page load, since it just zooms in from the top left.

Reply:
@byronbest I think we're getting a bit into the weeds here. What I'm asking is to remove the addition of `infoBox.animate` from this PR. I encourage you to make another PR for the `animate` where we can talk more

Reply:
My first complaint is that the infoBox jumps when it reaches the edge, without animate.
Why not add your request as another suggestion, to argue for jumping around?

Reply:
Why try to convince the user that there are more than one infoBox, when there is clearly only one? "Hover mouse over a marker to show the InfoBox".

Reply:
The current model, implementation details aside, is that there is an infobox for each plane that appears when you hover over it. This is consistent with the website and mobile apps, and with most UIs that have a similar capability (spelling checkers, alt-text for images on web pages, etc). Adding animation is surprising.

Reply:
How about animate only if already visible? That should make everyone happy.

Reply:
I think any animation change should be the subject of a different pull request, then we can discuss and experiment there.

Reply:
My PR already has your suggestion, plus animate if already visible.
Of course I tested before I suggested it, check it out.
Jumps to the marker, slides smoothly when reaches edge. Wish I'd done that first, and saved all this back and forth.